onnx/defs/rnn/defs.cc

```diff
@@ -161,22 +193,36 @@ custom implementation such as CuDNN.
 `WBb[iofc]` - W bias vectors for backward input, output, forget, and cell gates
 `RBb[iofc]` - R bias vectors for backward input, output, forget, and cell gates
 `PB[iof]`  - P peephole weight vector for backward input, output, and forget gates
-`tanh(X)` - hyperbolic tangent of X
-`sigmoid(X)` - 1 / (1 + e^-X)
 `H` - Hidden state
 `num_directions` - 2 if direction == bidirectional else 1
 
+Activation functions:
+  ReLU(x)                - max(0, x)
+  tanh(x)                - (1 - e^{-2x})/(1 + e^{-2x})
+  sigmoid(x)             - 1/(1 + e^{-x})
+  (NOTE: Below are optional)
+  linear(x)              - alpha*x + beta
+  leakyReLU(x)           - x if x >= 0 else alpha * x
+  thresholdedReLU(x)     - x if x >= alpha else 0
+  PReLU(xi)               - xi if xi >= 0 else alpha[i]* xi over dim 0
+  scaledTanh(x)          - alpha*tanh(beta*x)
+  sigmoidHard(x)         - min(max(alpha*x + beta, 0), 1)
+  ELU(x)                 - x if x >= 0 else alpha*(e^x - 1)
+  softsign(x)            - x/(1 + |x|)
+  softplus(x)            - log(1 + e^x)
+  parametricSoftplus(xi) - alpha[i]*log(1 + e^{beta[i]* xi}) over dim 0
+
 Equations (forward LSTM with default activations and peepholes):
-  - it = sigmoid(Wi*Xt + Ri*Ht-1 + Pi (.) Ct-1 + Wbi + Rbi)
-  - ft = sigmoid(Wf*Xt + Rf*Ht-1 + Pf (.) Ct-1 + Wbf + Rbf)
-  - ct = tanh(Wc*Xt + Rc*Ht-1 + Wbc + Rbc)
+  - it = sigmoid((Xt^T)*Wi + Ht-1*Ri + Pi (.) Ct-1 + Wbi + Rbi)
+  - ft = sigmoid((Xt^T)*Wf + Ht-1*Rf + Pf (.) Ct-1 + Wbf + Rbf)
+  - ct = tanh((Xt^T)*Wc + Ht-1*Rc + Wbc + Rbc)
   - Ct = ft (.) Ct-1 + it (.) ct
-  - ot = sigmoid(Wo*Xt + Ro*Ht-1 + Po (.) Ct + Wbo + Rbo)
-  - H = ot (.) tanh(Ct)
+  - ot = sigmoid((Xt^T)*Wo + Ht-1*Ro + Po (.) Ct + Wbo + Rbo)
+  - Ht = ot (.) tanh(Ct)
 )DOC")
-    .Attr("activations", "A list of 4 (or 8 if bidirectional) activation functions "
-          "for input, output, forget, and cell gates. The activation functions must "
-          "be one of sigmoid and tanh. See the equations for default.",
+    .Attr("activations", "A list of 3 (or 6 if bidirectional) activation functions "
```

Comment:
I'd say that this whole argument is optional and implementations don't have to support it (including overriding standard activations with tanh/sigmoid.

Reply:
Yes, it is optional. I will make it more explicit.